src/jit.jl

```diff
@@ -460,6 +469,10 @@ function cufunction(dev::CuDevice, @nospecialize(func), @nospecialize(tt))
         error("Device capability v$dev_cap not supported by available toolchain")
     cap = maximum(compat_caps)
 
+    if compile_hook[] != nothing
+        compile_hook[](func, tt, cap)
```

Comment:
For similar hooks in Base I have to use `invokelatest`, maybe not necessary here since we set it to a new lambda everytime.